### ISSUE TITLE
DB2 connection error in 6.1.2.201907061627

### ISSUE THREAD

**Issue by Bosco2k6** [dbeaver/dbeaver#6249]
#### System information: 
- Operating system: Windows 10 Pro  Version 1903
- DBeaver version:  6.1.2.201907061627

#### Connection specification:
- Database name and version:   DB2  LUW  Version 11.1

#### Describe your question:
I'm getting the following error when I try to connect to DB: 

`[jcc][t4][10262][11223][4.21.29] Unexpected Throwable caught: null. ERRORCODE=-4228, SQLSTATE=null`

I've Googled and see info about using the wrong version of the "JCE" for the Oracle JRE - if that's the issue, I'm not sure how to fix it. 

Can someone help/please advise ?   LMK - thanks!

**Comment by Sargul** [comment]
Does the DB2 Server use encryption?
If you are the DBA check DBM CFG for AUTHENTICATION and SSL_SVR_KEYDB parameter.
SSL_SVR_KEYDB indicates that SSL is enabled. 
AUTHENTICATION set to anything else as SERVER: you have to specify the correct securityMechanism property for the jdbc connection:

https://querysurge.zendesk.com/hc/en-us/articles/206083403-Configuring-Connections-IBM-DB2-With-Security-Mechanism

**Comment by Bosco2k6** [comment]
Thanks for the reply / information. It looks like I have solved this, and the error/exception was a bit misleading. 

I took a step back and used the `C:\...\SQLLIB\BIN\db2val.exe` to validate my installation. It had some complaints, and I think the most significant was that my local user (admin) was not in the Windows DB2ADMNS/DB2USERS group. I added the user, and also used the "Repair" option of 'Add or Remove' software for DB2 to let it copy/correct some allegedly missing files. 

After those steps `db2val.exe` had 0 complaints, and now the connection is working fine. Thanks again for trying to help! 

ps. For the record, `db2val` command output looks like this when it is happy: 

`DBI1379I  The db2val command is running. This can take several minutes.`

`DBI1333I  Installation file validation for the DB2 copy DB2COPY1
      was successful.`

`DBI1337I  The partitioned database environment validation for the
      instance DB2 was successful.`

`DBI1339I  The instance validation for the instance DB2 was
      successful.`

`DBI1343I  The db2val command completed successfully. For details, see
      the log file C:\Users\admin\DOCUME~1\DB2LOG\db2val-Wed Jul 10 07_55_03 2019.log.`
